coder-sdk/client.go

```diff
@@ -1,46 +1,265 @@
+
+	// SiteSetupModeEnabled fetches the current setup_mode state of a Coder Enterprise deployment.
+	SiteSetupModeEnabled(ctx context.Context) (bool, error)
+
+	// SiteConfigExtensionMarketplace fetches the extension marketplace configuration.
+	SiteConfigExtensionMarketplace(ctx context.Context) (*ConfigExtensionMarketplace, error)
+
+	// PutSiteConfigExtensionMarketplace sets the extension marketplace configuration.
+	PutSiteConfigExtensionMarketplace(ctx context.Context, req ConfigExtensionMarketplace) error
+
+	// DeleteDevURL deletes the specified devurl.
+	DeleteDevURL(ctx context.Context, envID, urlID string) error
+
+	// CreateDevURL inserts a new devurl for the authenticated user.
+	CreateDevURL(ctx context.Context, envID string, req CreateDevURLReq) error
+
+	// DevURLs fetches the Dev URLs for a given environment.
+	DevURLs(ctx context.Context, envID string) ([]DevURL, error)
+
+	// PutDevURL updates an existing devurl for the authenticated user.
+	PutDevURL(ctx context.Context, envID, urlID string, req PutDevURLReq) error
+
+	// CreateEnvironment sends a request to create an environment.
+	CreateEnvironment(ctx context.Context, req CreateEnvironmentRequest) (*Environment, error)
+
+	// ParseTemplate parses a template config. It support both remote repositories and local files.
+	// If a local file is specified then all other values in the request are ignored.
+	ParseTemplate(ctx context.Context, req ParseTemplateRequest) (Template, error)
+
+	// CreateEnvironmentFromRepo sends a request to create an environment from a repository.
+	CreateEnvironmentFromRepo(ctx context.Context, orgID string, req Template) (*Environment, error)
+
+	// Environments lists environments returned by the given filter.
+	Environments(ctx context.Context) ([]Environment, error)
+
+	// UserEnvironmentsByOrganization gets the list of environments owned by the given user.
+	UserEnvironmentsByOrganization(ctx context.Context, userID, orgID string) ([]Environment, error)
+
+	// DeleteEnvironment deletes the environment.
+	DeleteEnvironment(ctx context.Context, envID string) error
+
+	// StopEnvironment stops the environment.
+	StopEnvironment(ctx context.Context, envID string) error
+
+	// RebuildEnvironment requests that the given envID is rebuilt with no changes to its specification.
+	RebuildEnvironment(ctx context.Context, envID string) error
+
+	// EditEnvironment modifies the environment specification and initiates a rebuild.
+	EditEnvironment(ctx context.Context, envID string, req UpdateEnvironmentReq) error
+
+	// DialWsep dials an environments command execution interface
+	// See https://github.com/cdr/wsep for details.
+	DialWsep(ctx context.Context, baseURL *url.URL, envID string) (*websocket.Conn, error)
+
+	// DialExecutor gives a remote execution interface for performing commands inside an environment.
+	DialExecutor(ctx context.Context, baseURL *url.URL, envID string) (wsep.Execer, error)
```

Comment:
is wsep.Execer still available since the repo was archived? 

url.URL also seems like an interesting API choice.

Reply:
I dunno, I didn't really look at any of these APIs, I just extracted the public methods into an interface. Is there something we should switch over to?

Reply:
ah whoops, probably need to unarchive the wsep repo for this reason.